lib/sse_consumer.ex

```diff
@@ -0,0 +1,150 @@
+defmodule SSEConsumer do
+  require Logger
+  use GenServer
+
+  defmodule Request do
+    @enforce_keys ~w(method url body headers)a
+    defstruct @enforce_keys
+  end
+
+  defmodule State do
+    @moduledoc false
+    @enforce_keys ~w(recipient request conn_id remaining_stream)a
+    defstruct @enforce_keys
+
+    def new(recipient, request = %SSEConsumer.Request{}) when is_pid(recipient) do
+      %__MODULE__{
+        recipient: recipient,
+        request: request,
+        remaining_stream: "",
+        conn_id: nil
+      }
+    end
+
+
+    def set_connection(%__MODULE__{conn_id: current_conn_id} = state, new_conn_id)
+    when new_conn_id != nil and is_nil(current_conn_id) do
+      %{state | conn_id: new_conn_id}
+    end
+  end
+
+  #===========================================================================
+  # Public API
+  #===========================================================================
+
+  def stream_to(recipient, request, options), do: start_link(recipient, request, options)
```

Comment:
because `stream_to` is separate to `start_link` does that mean that stream_to could be called multiple times on the same consumer process?

Reply:
oh sorry being a fool. this calls start link